Carbon.Redis/Abstractions/ICarbonCacheExtensions.cs

```diff
@@ -83,39 +127,164 @@ public static async Task HashSetAsObject<T>(this ICarbonCache instance, string k
             return JsonSerializer.Deserialize<T>(JsonSerializer.Serialize(keyValuePairs));
         }
 
+        /// <summary>
+        /// Gets your entire object with the given fields that is converted into hash key-value pairs within a single redis hash key, if key does not exist, it tries to fill with the given function
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="instance"></param>
+        /// <param name="key">Redis Key</param>
+        /// <param name="setMethodIfNotExists">Execute if this key does not exist</param>
```

Comment:
Can we use IfNotFound or IfDoesNotExist instead of IfNotExists, in the name of grammatical correctness?

Reply:
I hate using Do or Does in variable names haha. It is indeed inspired from SQL's **IF EXISTS** or **IF NOT EXISTS** phrases.